metrics/ci_integrations/docs/diagrams/ci_integrations_config_validator_destination_class_diagram.puml

```diff
@@ -0,0 +1,86 @@
+@startuml ci_integrations_config_validator_class_diagram
+
+    package integration.interface {
+        package destination.config as interface_config {
+
+            package validation_delegate as interface_delegate {
+                interface DestinationConfigValidationDelegate {}
+            }
+
+            package validator_factory as interface_factory {
+                interface ConfigValidatorFactory {}
+            }
+
+            package model as interface_model {
+                abstract class DestinationConfigValidationResult {}
+                
+                abstract class DestinationConfigValidationResultBuilder {}
```

Comment:
Should we move this class into a `builder` package? The same suggestion for `CoolIntegrationDestinationConfigValidationResultBuilder` class.